tests/API/AuthApiTest.php

```diff
@@ -2,31 +2,80 @@
+use Auth0\SDK\API\Helpers\InformationHeaders;
 use Auth0\Tests\API\ApiTests;
 
 class AuthApiTest extends ApiTests
 {
 
-    public function testAuthorize()
+    public static $telemetry;
+
+    public static $telemetryParam;
+
+    public static function setUpBeforeClass()
     {
-        $domain    = 'dummy.auth0.com';
-        $client_id = '123456';
+        $infoHeadersData = new InformationHeaders;
+        $infoHeadersData->setCorePackage();
 
-        $api = new Authentication($domain, $client_id);
+        self::$telemetry      = urlencode( $infoHeadersData->build() );
+        self::$telemetryParam = 'auth0Client='.self::$telemetry;
+    }
 
-        $authorize_url = $api->get_authorize_link('code', 'http://lala.com');
+    public function testThatBasicAuthorizeLinkIsBuiltCorrectly()
+    {
+        $api = new Authentication('test-domain.auth0.com', '__test_client_id__');
+
+        $authorize_url       = $api->get_authorize_link('code', 'https://example.com/cb');
+        $authorize_url_parts = parse_url( $authorize_url );
+
+        $this->assertEquals('https', $authorize_url_parts['scheme']);
+        $this->assertEquals('test-domain.auth0.com', $authorize_url_parts['host']);
+        $this->assertEquals('/authorize', $authorize_url_parts['path']);
+
+        $authorize_url_query = explode( '&', $authorize_url_parts['query'] );
+        $this->assertContains('redirect_uri=https%3A%2F%2Fexample.com%2Fcb', $authorize_url_query);
+        $this->assertContains('response_type=code', $authorize_url_query);
+        $this->assertContains('client_id=__test_client_id__', $authorize_url_query);
+        $this->assertContains(self::$telemetryParam, $authorize_url_query);
+        $this->assertNotContains('connection=', $authorize_url_parts['query']);
+        $this->assertNotContains('state=', $authorize_url_parts['query']);
+    }
 
-        $this->assertEquals(
-            'https://dummy.auth0.com/authorize?response_type=code&redirect_uri=http%3A%2F%2Flala.com&client_id=123456',
-            $authorize_url
-        );
+    public function testThatAuthorizeLinkIncludesConnection()
+    {
+        $api = new Authentication('test-domain.auth0.com', '__test_client_id__');
 
-        $authorize_url2 = $api->get_authorize_link('token', 'http://lala.com', 'facebook', 'dastate');
+        $authorize_url       = $api->get_authorize_link('code', 'https://example.com/cb', 'test-connection');
+        $authorize_url_query = parse_url( $authorize_url, PHP_URL_QUERY );
+        $authorize_url_query = explode( '&', $authorize_url_query );
 
-        $this->assertEquals(
-            'https://dummy.auth0.com/authorize?response_type=token&redirect_uri=http%3A%2F%2Flala.com'.'&client_id=123456&connection=facebook&state=dastate',
-            $authorize_url2
-        );
+        $this->assertContains('connection=test-connection', $authorize_url_query);
+        $this->assertContains(self::$telemetryParam, $authorize_url_query);
```

Comment:
I think the `telemetryParam` assert can either be in the `testThatBasicAuthorizeLinkIsBuiltCorrectly` or in a separate test for this purpose. No need to duplicate this for every property that can be changed.. (This comment applies to all the tests introduced here). Unless of course, you have a setter to remove (not send) the telemetry header. That would also be a separate test, IMO.

Reply:
Just making sure it's not clobbered. Yes, a bit of overkill here.